sfge/src/main/java/com/salesforce/graph/build/AbstractApexVertexBuilder.java

```diff
@@ -84,20 +105,44 @@ private void buildVertices(JorjeNode node, Vertex vNodeParam, String fileName) {
             if (StaticBlockUtil.isStaticBlockStatement(node, child)) {
                 final Vertex parentVertexForChild =
                         StaticBlockUtil.createSyntheticStaticBlockMethod(g, vNode, i);
+                // The child vertex is made a child of the synthetic vertex instead of its original
+                // parent.
                 GremlinVertexUtil.addParentChildRelationship(g, parentVertexForChild, vChild);
                 verticesAddressed.add(parentVertexForChild);
             } else {
                 GremlinVertexUtil.addParentChildRelationship(g, vNode, vChild);
             }
 
-            if (vPreviousSibling != null) {
-                g.addE(Schema.NEXT_SIBLING).from(vPreviousSibling).to(vChild).iterate();
-            }
-            vPreviousSibling = vChild;
+            // Map this vertex as the Nth child vertex of the parent.
+            // TODO: This might be a bug. It's possible we should actually be using the synthetic
+            //       vertex if we create one. Do more investigating around that!
+            childVerticesByIndex.put(child.getChildIndex(), vChild);
```

Comment:
I did this in order to preserve as much of the existing behavior around the `NEXT_SIBLING` edge as possible. I do believe this specific behavior is a bug, but I don't want to be messing with things beyond the specific scope of the bug in question right now.

Reply:
are you concerned about regression? do you think the tests in and around this area is sufficient to give you a confidence if you were to push the scope? or you just choosing to play safe for now?

Reply:
I'm not concerned about a regression. This is actually about preserving the existing behavior, even though I'm pretty sure that the existing behavior is connected to a known bug separate from the one being fixed here.